### PR TITLE
ISSUE-43

### LINKED ISSUE #43: ToSlice dose not re-empty the dist slice
Codes:

```go
func main() {
	l := []string{"foo", "bar", "baz"}
	linq.From(l).ToSlice(&l)
	fmt.Println(l)
}
```

Result:

```
[foo bar baz foo bar baz]
```

go-linq version:

```
- name: github.com/ahmetalpbalkan/go-linq
  version: 8985ec99e11a8bff7eb19dd0a0b2187770dab23a
```

I don't know whether this is a bug or a feature, but if this is a feature, I think it's better to have this behavior documented on `ToSlice`'s API doc.

### PR DESCRIPTION
**Pull request by kalaninja**
Adding docs on ToMap, ToMapBy and ToSlice methods. This resolves #43

### CODE REVIEW

**Comment by kalaninja** [comment]
@ahmetalpbalkan please, check

File: result.go [hunk 1]
```diff
@@ -356,10 +357,11 @@ func (q Query) ToMap(result interface{}) {
 		})
 }
 
-// ToMapBy iterates over a collection and populates result map with elements.
+// ToMapBy iterates over a collection and populates the result map with elements.
 // Functions keySelector and valueSelector are executed for each element of the collection
 // to generate key and value for the map. Generated key and value types must be assignable
 // to the map's key and value types.
+// ToMapBy doesn't empty the result map before populating it.
```
Comment (ahmetalpbalkan) [inline result.go]: if you want to make these new paragraphs, you need to have an empty comment line before the text.

**Comment by ahmetalpbalkan** [comment]
Looks good.

**Comment by kalaninja** [comment]
@ahmetalpbalkan  @aisk  So, if everyone's agree, I am merging this.

**Pull request merged**